### DISCUSSION TITLE
Allow an LR less than 1e-4 for Lora.

### DISCUSSION THREAD

**Discussion by JuanForge** [bghira/SimpleTuner#2661] [Ideas]
Hello,
I'm trying to train a LoRA with a lower learning rate to improve training quality, but SimpleTuner seems to ignore my configuration.

Even when I set a value like 5e-5, the backend still trains at 1e-4.

Everything looks correct in the intermediate configs, but in the UI I can clearly see the Learning Rate displayed as 1.00e-4.

Is there something overriding the value, or am I missing a parameter?

Latest version in the 'Release' branch (v4.1.2) for a LoRa Z Image
<img width="1327" height="212" alt="image" src="https://github.com/user-attachments/assets/6614901e-91da-4467-8436-72465f63000c" />
<img width="1262" height="350" alt="image" src="https://github.com/user-attachments/assets/53d9b746-f71d-43f8-9b13-8fe4284d5e1f" />

**Comment by JuanForge** [comment]
Additionally, I’m experiencing instability with the step loss during training.

The loss fluctuates significantly, which often results in newer checkpoints performing worse than earlier ones.

This makes it difficult to identify the best model without manually testing every version.

Do you have any recommendations to stabilize the training and reduce these fluctuations?

Thanks in advance.

**Comment by bghira** [comment]
you're likely resuming training and trying to change the learning rate. but the optimiser state resumes with the learning rate from the previous run. there's nothing actually preventing (in the codebase) a lower learning rate from being used. i think it's just the resume blocking it.

**Reply by JuanForge** [reply]
Hi,
I’ve done some testing and confirmed that the learning rate wasn’t updating because I resumed training from a checkpoint. Thanks for pointing that out.

Do you have any recommendations to stabilize the step loss? Right now it decreases but remains quite noisy, which forces me to manually test multiple checkpoints to find the best one.

For reference, here are some recent step loss values:
0.428, 0.269, 0.266, 0.298, 0.145, 0.328, 0.362, 0.427, 0.343, 0.331, 0.253, 0.569, ...
This instability is also noticeable during inference.

My current setup:

Batch size: 1
Gradient accumulation: 1 (also tested 4, 8, 12, 32 without significant improvement)
Learning rate: 9e-5 (constant)
Max gradient norm: 2.0

Sorry for the late reply.

( translated by: GPT ) 
<img width="1110" height="34" alt="image" src="https://github.com/user-attachments/assets/f3a9af2a-00a4-4a2a-9a10-b3a28af9858e" />
<img width="1115" height="27" alt="image" src="https://github.com/user-attachments/assets/4ec3deb4-c4bf-4a56-a252-bc42369b569c" />
